node/plugin/cmd/main.go

```diff
@@ -135,12 +136,36 @@ func pluginOps(ctx *cli.Context) {
 		return
 	}
 
-	_, dispersalPort, retrievalPort, v2DispersalPort, v2RetrievalPort, err := core.ParseOperatorSocket(config.Socket)
+	opSocket, err := core.ParseOperatorSocket(config.Socket)
 	if err != nil {
 		log.Printf("Error: failed to parse operator socket: %v", err)
 		return
 	}
 
+	_, dispersalPort, err := net.SplitHostPort(opSocket.GetV1DispersalSocket())
```

Comment:
Actually these look odd to me: with `ParseOperatorSocket` you already parsed the socket, but then you use `GetV1DispersalSocket` to encode it back to string, and then use `SplitHostPort` to decode again.

Reply:
We don't want to be exporting internal fields of the struct and have well defined methods to access variables. 
This allows us to have defensive coding mechanisms where we don't let pkg importee modify internal state of operator socket. 

As we want to have access to read only value of dispersal port, didn't want to make these exportable fields in the struct def.

Reply:
I think it's idiomatic to make fields public if they need to be accessed outside in golang. If you really want read-only struct (does it provide that much value here?), then adding getters should be fiine

Reply:
It adds value by not letting anybody outside modify endpoint state. Essentially the struct wraps network information. This should be abuse proof outside of the package instantiating it.

Reply:
I think it's minor. If you really want it, I'd add getters. It's convoluted to do this parse-encode-parse loop here.

Reply:
Unresolved: I don't see this is addressed

Reply:
Making the fields public when not necessary would be a code smell. It allows for unnecessary modification outside of package. I don't see an issue with keeping this as is.